pkg/apis/work/v1alpha2/well_known_constants.go

```diff
@@ -124,6 +124,12 @@ const (
 	// Additional options will be added here in the future.
 	DeletionProtectionLabelKey = "resourcetemplate.karmada.io/deletion-protected"
 	DeletionProtectionAlways   = "Always"
+
+	// ResourceBindingFailoverLabel If a resource is failed over by karmada, this label will be attached
+	// to the rescheduled workload. The value will denote the type of failover that occurred, either cluster or application.
+	// This can be useful if applications are stateful and need to know when they have been failed over by Karmada,
+	// as opposed to being scheduled fresh.
+	ResourceBindingFailoverLabel = "resourcebinding.karmada.io/failover-type"
```

Comment:
I'm not sure if we need this label yet. 
But given this PR focuses on the failover history, I suggest moving this out of this PR, and letting this PR focus on:
- failover history API design
- the logic of generating the history
- use the history to prevent scheduling failure clusters again

Reply:
That sounds fair to me. We currently use the label for our e2e failover feature, but I see no problem separating this out.

Reply:
Great!
The reason why I prefer small PRs because they usually focus on a small area and make it easier to make final decisions.

Reply:
Are we going to remove this label from this PR?

Reply:
Yes, I'll be pulling out the failover flag logic from this PR. :)

Reply:
I guess we can remove the definition as well from this PR.